File: backend/app/api/v1/endpoints/stats.py
"""CropGuard Stats Endpoints."""

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.api.deps import get_current_active_user, get_db
from app.models.user import User
from app.services import stats_service

router = APIRouter(prefix="/stats", tags=["Statistics"])


@router.get("/overview", summary="Get personal dashboard stats")
async def stats_overview(
    current_user: User = Depends(get_current_active_user),
    db: AsyncSession = Depends(get_db),
) -> dict:
    """Return aggregated stats for the current user's dashboard."""
    return await stats_service.get_user_overview(current_user.id, db)


@router.get("/outbreak", summary="Get geo-clustered outbreak data (public)")
async def outbreak_data(db: AsyncSession = Depends(get_db)) -> list:
    """Return geo-clustered disease predictions for the outbreak map."""
    return await stats_service.get_outbreak_data(db)

File: frontend/src/api/stats.ts
import apiClient from './client'
import type { StatsOverview } from '../types'

// ─── Stats API ──────────────────────────────────────────────────────────────

export const statsApi = {
  /**
   * GET /stats/overview
   * Returns aggregated stats for the current authenticated user's dashboard.
   */
  overview: async (): Promise<StatsOverview> => {
    const res = await apiClient.get<StatsOverview>('/stats/overview')
    return res.data
  },

  /**
   * GET /stats/outbreak
   * Returns geo-clustered disease predictions for the outbreak map (public).
   */
  outbreak: async (): Promise<unknown[]> => {
    const res = await apiClient.get<unknown[]>('/stats/outbreak')
    return res.data
  },
}
